agent/engine/docker_task_engine.go

```diff
@@ -830,6 +832,16 @@ func (engine *DockerTaskEngine) Capabilities() []string {
 		}
 	}
 
+	// TODO we also need to disable this if agent already managing older tasks
+	if engine.cfg.TaskCPUMemLimit {
+		if _, ok := supportedVersions[dockerclient.Version_1_22]; ok {
+			capabilities = append(capabilities, capabilityPrefix+capabilityTaskCPUMemLimit)
+		} else {
+			seelog.Warn("Task CPU + Mem Limit disabled due to unsupported Docker version")
+			engine.cfg.TaskCPUMemLimit = false
```

Comment:
The approach we've taken recently is to kill the agent with a terminal error code if it can't register with the required capability when asked to do so via the config. We should rethink if we have to take the same approach here. The argument is that it's better to not register the instance than register it without the capability that was asked for as it leads to a more consistent troubleshooting experience.

Reply:
I think its slightly different here. In this case, 'enabled' is the default and it can be manually overridden to false.

Since it is a default, it would be disruptive to not register the instance (in case of updates, old versions of docker, etc).

Reply:
After looking over some things further we think that the best way to handle this is to make enhancements to the config package. We need to fail if the value was set explicitly, but we should only fall back to old behavior gracefully if the value was derived from a default. 

Its a significant change, so I'd like to do that on a separate PR and keep progress moving forward on this branch.